packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -75,3 +101,65 @@ class CoverageCollector {
     return await formatter.format(_globalHitmap);
   }
 }
+
+/// A class that represents a pending task of coverage data collection.
+/// Instances of this class are obtained when a process starts running code
+/// by calling [CoverageCollector.addTask]. Then, when the code has run to
+/// completion (all the coverage data has been recorded), the task is started
+/// to actually collect the coverage data.
+class CoverageCollectionTask {
+  final Completer<Null> _completer = new Completer<Null>();
+  final CoverageCollector _collector;
+  final String _host;
+  final int _port;
+  final Process _processToKill;
+
+  CoverageCollectionTask(
```

Comment:
it may be safer to give this a private constructor, but i don't feel strongly about this.